packages/vm/src/instance.rs

```diff
@@ -213,11 +216,15 @@ where
             }
         }
 
+        let lock = m.map(|m| m.lock().unwrap());
```

Comment:
Does this ever create a lock? `m: Option<&Mutex<()>>` seems to be set to `None` in all caller code. Then the callback in `map` is never executed and `lock` is always `None`.

Reply:
Except for the call that interests us. See cache.rs, line 266 above.

Reply:
Oh, right, missed that one.

Reply:
We can turn this into a mandatory mutex, but I think it's more flexible this way.

And of course, I think we should create an issue in wasmer, to try and find the root cause / fix this for good.